src/lenses/CrvRefundLens.sol

```diff
@@ -0,0 +1,153 @@
+    }
+
+    // Get MIM gauge power of a voter. Power represents % of veCrv balance applied to gauge.
+    function getVoterMimGaugePower(address votingAddress) public view returns (uint256) {
+        (, uint256 power, ) = gaugeController.vote_user_slopes(votingAddress, CURVE_MIM_GAUGE_ADDR);
+        return power;
+    }
+
+    function getSpellPrice() public view returns (uint256) {
+        IOracle oracle = IOracle(SPELL_ORACLE_ADDR);
+        bytes memory data = abi.encodePacked(uint256(0));
+        return PRECISION ** 2 / oracle.peekSpot(data);
+    }
+
+    function getTotalMimGaugeVotes() public view returns (uint256) {
+        return gaugeController.get_gauge_weight(CURVE_MIM_GAUGE_ADDR);
+    }
+
+    function getWeeklySpellBribes() public view returns (uint256) {
+        uint256 rewardsPerGauge = yBribeContract.reward_per_gauge(CURVE_MIM_GAUGE_ADDR, SPELL_ADDR);
+        uint256 claimsPerGauge = yBribeContract.claims_per_gauge(CURVE_MIM_GAUGE_ADDR, SPELL_ADDR);
+        return rewardsPerGauge - claimsPerGauge;
+    }
+
+    function getVoterSpellBribes(address votingAddress) public view returns (uint256) {
+        uint256 totalMimGaugeVotes = getTotalMimGaugeVotes();
+        uint256 voterMimGaugeVotes = getVoterMimGaugeVotes(votingAddress);
+        uint256 weeklySpellBribes = getWeeklySpellBribes();
+
+        // Pro-rate total SPELL bribes by voter's share of gauge votes
+        return (weeklySpellBribes * voterMimGaugeVotes) / totalMimGaugeVotes;
+    }
+
+    function getVoterSpellBribesUsd(address votingAddress) public view returns (uint256) {
+        return (getVoterSpellBribes(votingAddress) * getSpellPrice()) / PRECISION;
+    }
+
+    function getRefundInfo(ICauldronV4[] calldata cauldrons, address user, address votingAddress) public view returns (RefundInfo memory) {
+        address[] memory cauldronContracts = new address[](cauldrons.length);
+        for (uint256 i = 0; i < cauldrons.length; i++) {
+            cauldronContracts[i] = address(cauldrons[i]);
+        }
+
+        return
+            RefundInfo({
+                cauldrons: cauldronContracts,
+                spellPrice: getSpellPrice(),
+                userBorrowAmounts: getUserBorrowAmounts(cauldrons, user),
+                userVeCrvVoted: getVoterMimGaugeVotes(votingAddress),
+                userBribesReceived: getVoterSpellBribesUsd(votingAddress)
+            });
+    }
+
+    function handleFees(ICauldronV4[] calldata cauldrons, uint128 refund) external returns (uint128 totalFeesWithdrawn) {
+        totalFeesWithdrawn = 0;
+
+        for (uint256 i = 0; i < cauldrons.length; i++) {
+            ICauldronV4 cauldron = cauldrons[i];
+
+            Rebase memory totalBorrow = cauldron.totalBorrow();
+            (uint64 lastAccrued, uint128 feesEarned, uint64 INTEREST_PER_SECOND) = cauldron.accrueInfo();
+            uint256 elapsedTime = block.timestamp - lastAccrued;
+
+            console2.log("block.timestamp", block.timestamp);
```

Comment:
be sure to remove the console log when deploying (that should trapped by the hardhat task script tho)